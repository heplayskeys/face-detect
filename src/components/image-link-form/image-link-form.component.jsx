import React from 'react';
import { connect } from 'react-redux';
import { setInputField } from '../../redux/input/input.actions';
import FaceCount from '../face-count/face-count.component.jsx';
import './image-link-form.styles.scss';

const ImageLinkForm = props => {
	const { input, handleSubmit, numFaces, recentGrabs, setInput } = props;

	return (
		<div>
			{numFaces || recentGrabs.length ? (
				<div>
					<FaceCount numFaces={numFaces} />
				</div>
			) : (
				<p className='f4 mb4'>{'Grab and count the faces in your image!'}</p>
			)}
			<div className='center'>
				<div
					id='input-container'
					className='pa4 br3 shadow-5 center form justify-around'
				>
					<input
						className='user-input f4 pa2 w-75-ns'
						type='text'
						placeholder='input image URL here'
						onChange={setInput}
						value={input}
					/>
					<button
						className='grab-btn w-20-ns f4 grow link ph3 pv2 dib white mybg'
						onClick={handleSubmit}
					>
						{'grab'}
					</button>
				</div>
			</div>
		</div>
	);
};

const mapDispatchToProps = dispatch => ({
	setInput: e => dispatch(setInputField(e.target.value))
});

export default connect(null, mapDispatchToProps)(ImageLinkForm);
