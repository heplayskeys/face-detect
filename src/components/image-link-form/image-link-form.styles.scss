$breakpoint-tablet: 768px;

#input-container {
	justify-content: space-around;

	@media screen and (max-width: $breakpoint-tablet) {
		width: 100%;
	}
}

.form {
	width: 700px;
}

.mybg {
	background: rgb(2, 142, 191);
	font-weight: bold;

	&:active {
		font-weight: normal;
	}
}
