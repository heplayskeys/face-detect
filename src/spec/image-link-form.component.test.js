import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { ImageLinkForm } from '../components/image-link-form/image-link-form.component';

let wrapper;

describe('<ImageLinkForm />', () => {
	beforeEach(() => {
		const mockProps = {
			input: '',
			handleSubmit: jest.fn(),
			numFaces: 0,
			recentGrabs: [],
			setInput: jest.fn()
		};

		wrapper = shallow(<ImageLinkForm {...mockProps} />);
	});

	it('renders image link form without face detect component', () => {
		expect(shallowToJson(wrapper)).toMatchSnapshot();
	});

	it('renders image link form with face detect component', () => {
		wrapper.setProps({ numFaces: 10 });
		expect(shallowToJson(wrapper)).toMatchSnapshot();
	});
});
