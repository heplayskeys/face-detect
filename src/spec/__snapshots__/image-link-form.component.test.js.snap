// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ImageLinkForm /> renders image link form with face detect component 1`] = `
<div>
  <div>
    <FaceCount
      numFaces={10}
    />
  </div>
  <div
    className="center"
  >
    <div
      className="br3 shadow-5 center form justify-around"
      id="input-container"
      style={
        Object {
          "padding": "1.5rem",
        }
      }
    >
      <input
        className="user-input pa2 f4 w-75-ns"
        onChange={[Function]}
        placeholder="input image URL here"
        type="text"
        value=""
      />
      <button
        className="grab-btn w-20-ns f4 grow link ph3 pv2 dib white mybg"
        onClick={[MockFunction]}
      >
        grab
      </button>
    </div>
  </div>
  <div
    className="center"
  >
    <div
      className="br3 shadow-5 center form justify-around"
      id="input-container"
      style={
        Object {
          "padding": "0.75rem",
        }
      }
    >
      <form
        id="imageForm"
        style={
          Object {
            "width": "100%",
          }
        }
      >
        <input
          accept="image/*"
          className="user-file-input pa2 f4 w-75-ns"
          id="file-upload-input"
          onChange={[Function]}
          style={
            Object {
              "overflow": "hidden",
              "width": "100%",
            }
          }
          type="file"
        />
        <button
          className="upload-btn w-20-ns f4 grow link ph3 pv2 dib white mybg"
          onClick={[Function]}
        >
          upload
        </button>
      </form>
    </div>
  </div>
</div>
`;

exports[`<ImageLinkForm /> renders image link form without face detect component 1`] = `
<div>
  <p
    className="f4 mt3 mb3"
  >
    Grab and count the faces in your image!
  </p>
  <div
    className="center"
  >
    <div
      className="br3 shadow-5 center form justify-around"
      id="input-container"
      style={
        Object {
          "padding": "1.5rem",
        }
      }
    >
      <input
        className="user-input pa2 f4 w-75-ns"
        onChange={[Function]}
        placeholder="input image URL here"
        type="text"
        value=""
      />
      <button
        className="grab-btn w-20-ns f4 grow link ph3 pv2 dib white mybg"
        onClick={[MockFunction]}
      >
        grab
      </button>
    </div>
  </div>
  <div
    className="center"
  >
    <div
      className="br3 shadow-5 center form justify-around"
      id="input-container"
      style={
        Object {
          "padding": "0.75rem",
        }
      }
    >
      <form
        id="imageForm"
        style={
          Object {
            "width": "100%",
          }
        }
      >
        <input
          accept="image/*"
          className="user-file-input pa2 f4 w-75-ns"
          id="file-upload-input"
          onChange={[Function]}
          style={
            Object {
              "overflow": "hidden",
              "width": "100%",
            }
          }
          type="file"
        />
        <button
          className="upload-btn w-20-ns f4 grow link ph3 pv2 dib white mybg"
          onClick={[Function]}
        >
          upload
        </button>
      </form>
    </div>
  </div>
</div>
`;
